### PR TITLE
Document online consensus

### PR DESCRIPTION
**Pull request by jfbastien**
At the last CG meeting I was asked to make sure that we could make forward progress online. We also intend to hold semi-frequent video chats for the CG. Folks were concerned that the consensus process we documented for in-person meetings was too onerous in-between meetings, I'm therefore proposing that we adopt this online process. Talking to our standards folks this is usually left undocumented, but they like that this is precise and therefore unambiguous.

### CODE REVIEW

File: process/consensus.md [hunk 1]
```diff
@@ -40,6 +42,62 @@ In some cases where consensus seems to obviously have been reached, the chair
 may poll a lighter-weight request for objections to unanimous consensus. A
 single objection is sufficient to force the above consensus process.
 
+## Online consensus
+
+It it critical that work progress between in-person meetings: agreed-upon
```
Comment (lukewagner) [inline process/consensus.md]: *is

File: process/consensus.md [hunk 2]
```diff
@@ -40,6 +42,62 @@ In some cases where consensus seems to obviously have been reached, the chair
...
+designs need to move forward, and new ideas need to reach some level of maturity
+before being discussed in-person. To that end, this group can reach consensus
+online, either on GitHub repositories under the WebAssembly organization or in
+official video calls. In the later case, decisions are recorded in meeting notes
```
Comment (lukewagner) [inline process/consensus.md]: *latter

File: process/consensus.md (hunk 1 above)
Comment (MikeHolman) [inline process/consensus.md]: progresses?

File: process/consensus.md (hunk 2 above)
Comment (MikeHolman) [inline process/consensus.md]: do we really want to declare consensus on these calls? if so, how is consensus determined?

File: process/consensus.md [hunk 3]
```diff
@@ -40,6 +42,62 @@ In some cases where consensus seems to obviously have been reached, the chair
...
+   request that final consensus be delayed to an in-person meeting. The chair
+   can, at their discretion, refuse such a request.
+3. Substantial technical changes or additions are usually carried in their own
+   GitHub repository by a champion. It is critical that these proposals be able
+   to evolve quickly without much process. Early on in such a proposal's
+   lifetime no consensus is needed and a single champion can modify the proposal
+   at will. As the proposal matures it is expected that the champion will seek
+   collaborators to form a small group. Gauging consensus in the small group is
+   left up to the champion, with input from the chair. When a proposal is near
+   maturity to champion shall bring it to an in-person meeting and seek wider
```
Comment (MikeHolman) [inline process/consensus.md]: *the

File: process/consensus.md (hunk 2 above)
Comment (jfbastien) [inline process/consensus.md]: That's what some folks at the in-person meeting were asking for, yes. I tried to detail that process below, but will update it a bit.

**Comment by rossberg-chromium** [comment]
Sounds good, I agree it's important to make sure that we can make progress outside meetings, especially for people that can't come to every meeting.

It would be good to also think about the dual: that people who can't attend every meeting (because of limited time, budget or carbon footprint resources) get sufficient means of participating in decisions made at meetings. Seems especially relevant given the high frequency of meetings we seem to be heading for -- which, ironically, was decided at a meeting where not all parties were present.

**Comment by jfbastien** [comment]
> It would be good to also think about the dual: that people who can't attend every meeting (because of limited time, budget or carbon footprint resources) get sufficient means of participating in decisions made at meetings. Seems especially relevant given the high frequency of meetings we seem to be heading for -- which, ironically, was a decision made at a meeting where not all parties were present. ;)

The timing for the Kirkland meeting was set before the in-person meeting in Mountain View. It was merely on the agenda as an FYI.

**Comment by rossberg-chromium** [comment]
>
> The timing for the Kirkland meeting was set before the in-person meeting
> in Mountain View. It was merely on the agenda as an FYI.
>
Yes, I'm referring to the plan of having quarterly meetings in general,
which I was told was decided at the May meeting?

**Comment by jfbastien** [comment]
>> The timing for the Kirkland meeting was set before the in-person meeting
>> in Mountain View. It was merely on the agenda as an FYI.
>>
> Yes, I'm referring to the plan of having quarterly meetings in general,
which I was told was decided at the May meeting?

According to [the meeting notes](https://github.com/WebAssembly/meetings/blob/master/2017/CG-05.md) no such decision was made.

File: process/consensus.md [hunk 4]
```diff
@@ -40,6 +49,63 @@ In some cases where consensus seems to obviously have been reached, the chair
...
+   collaborators to form a small group. Gauging consensus in the small group is
+   left up to the champion, with input from the chair. When a proposal is near
+   maturity the champion shall bring it to an in-person meeting and seek wider
+   consensus on open design points and contended issues. All decisions made by
+   the small group can be revisited until consensus is reached at an in-person
+   meeting.
+
+Only 1. and 2. apply to the Working Group since the Community Group is the sole
+venue where substantial work can occur. It is expected that Working Group
+changes or additions in 2.'s category are validated by the Community Group
```
Comment (flagxor) [inline process/consensus.md]: -> substantial revision can occur?

Comment (jfbastien) [inline process/consensus.md]: The CG does original work as well, so it's not just revisions.

**Comment by jfbastien** [comment]
3 lgtms and some feedback from W3C. I'll merge for now and we'll tune the process after a bit of experience.

**Pull request merged**